ecosystem/python/sdk/aptos_sdk/secp256k1_ecdsa.py

```diff
@@ -0,0 +1,224 @@
+# Copyright © Aptos Foundation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import hashlib
+import unittest
+from typing import cast
+
+from ecdsa import SECP256k1, SigningKey, VerifyingKey
+
+from . import asymmetric_crypto
+from .bcs import Deserializer, Serializer
+
+
+class PrivateKey(asymmetric_crypto.PrivateKey):
+    LENGTH: int = 32
+
+    key: SigningKey
+
+    def __init__(self, key: SigningKey):
+        self.key = key
+
+    def __eq__(self, other: object):
+        if not isinstance(other, PrivateKey):
+            return NotImplemented
+        return self.key == other.key
+
+    def __str__(self):
+        return self.hex()
+
+    @staticmethod
+    def from_str(value: str) -> PrivateKey:
+        if value[0:2] == "0x":
+            value = value[2:]
+        if len(value) != PrivateKey.LENGTH * 2:
+            raise Exception("Length mismatch")
+        return PrivateKey(
+            SigningKey.from_string(bytes.fromhex(value), SECP256k1, hashlib.sha3_256)
+        )
+
+    def hex(self) -> str:
+        return f"0x{self.key.to_string().hex()}"
+
+    def public_key(self) -> PublicKey:
+        return PublicKey(self.key.verifying_key)
+
+    @staticmethod
+    def random() -> PrivateKey:
+        return PrivateKey(SigningKey.generate(curve=SECP256k1, hashfunc=hashlib.sha3_256))
+
+    def sign(self, data: bytes) -> Signature:
+        return Signature(self.key.sign(data))
```

Comment:
ohh, so we don't have to take the `sha3_256()` of the input data?

Reply:
it looks like it's set in the signing key by default as the hash function, see anywhere a PrivateKey instance is made

```python
    return PrivateKey(SigningKey.generate(curve=SECP256k1, hashfunc=hashlib.sha3_256))
```

Reply:
And public key